racket/src/ChezScheme/c/atomic.h

```diff
@@ -97,7 +97,7 @@ FORCEINLINE int S_cas_any_fence(int load_acquire, volatile void *addr, void *old
 # define CAS_LOAD_ACQUIRE(a, old, new) S_cas_any_fence(1, a, old, new)
 # define CAS_STORE_RELEASE(a, old, new) S_cas_any_fence(0, a, old, new)
 #elif (__GNUC__ >= 5) || defined(__clang__)
-# define CAS_ANY_FENCE(a, old, new) __sync_bool_compare_and_swap(a, old, new)
+# define CAS_ANY_FENCE(a, old, new) __sync_bool_compare_and_swap((uptr *)a, (uptr)old, (uptr)new)
```

Comment:
Either `uptr` or `ptr` works. But since this is a C macro, it's best to have parentheses around `a`, `old`, and `new`, so I'll merge a variant that does that.